protocol/x/vault/keeper/msg_server_deposit_to_vault.go

```diff
@@ -32,13 +34,16 @@ func (k msgServer) DepositToVault(
 	// Transfer from sender subaccount to vault.
 	// Note: Transfer should take place after minting shares for
 	// shares calculation to be correct.
+	if !quoteQuantums.IsUint64() {
```

Comment:
This check should be in `ValidateBasic` and not in the processing of the message.

Reply:
I first added it to `ValidateBasic` but moved it here to be clearer on why we require it to be `Uint64`. Can move back though